hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetUtils.java

```diff
@@ -71,35 +72,39 @@ public static int locationToDepth(String location) {
    *  Remove node from mutableExcludedNodes if it's covered by excludedScope.
    *  Please noted that mutableExcludedNodes content might be changed after the
    *  function call.
-   * @return the new excludedScope
    */
-  public static String removeDuplicate(NetworkTopology topology,
-      Collection<Node> mutableExcludedNodes, String excludedScope,
+  public static void removeDuplicate(NetworkTopology topology,
+      Collection<Node> mutableExcludedNodes, List<String> mutableExcludedScopes,
       int ancestorGen) {
     if (mutableExcludedNodes == null || mutableExcludedNodes.size() == 0 ||
```

Comment:
NIT: We can use CollectionUtils.isEmpty() to simplify the code at 79/80.

Reply:
OK